app/src/main/java/com/github/se/eventradar/ui/component/ReusableComponents.kt

```diff
@@ -144,162 +146,170 @@ fun EventCard(event: Event, onCardClick: (String) -> Unit) {
-
         // Buttons for category selection
         Text(
-            text = "Category: ",
-            modifier = Modifier.weight(1f),
+            text = stringResource(id = R.string.category_label),
+            modifier = Modifier.weight(1f).testTag("categoryLabel"),
             style =
                 TextStyle(
                     fontSize = 16.sp,
                 ))
 
         Row(
             verticalAlignment = Alignment.CenterVertically,
-            horizontalArrangement = Arrangement.Start) {
-              CategorySelection()
-            }
+            horizontalArrangement = Arrangement.Start,
+            modifier = Modifier.testTag("categoryRow"),
+        ) {
+          CategorySelection(
+              uiState = uiState,
+              modifier = Modifier.testTag("categoryOptionsColumn"),
+          )
+        }
 
         Spacer(modifier = Modifier.height(4.dp))
 
         // Button to apply filter
-        Row(modifier = Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.Center) {
-          Button(
-              onClick = {
-                // Apply filter
-                onRadiusChange(uiState.radiusInputFilter)
-                onFreeSelectionChange(uiState.freeEventsFilter)
-                onCategorySelectionChange(uiState.categorySelectionFilter)
-                onCategorySelectionChange(uiState.categorySelectionFilter)
-              }) {
-                Text("Apply")
-              }
-        }
+        Row(
+            modifier = Modifier.fillMaxWidth().testTag("filterApplyRow"),
+            horizontalArrangement = Arrangement.Center) {
+              Button(
+                  onClick = { onFilterApply() }, modifier = Modifier.testTag("filterApplyButton")) {
+                    Text(stringResource(id = R.string.filter_apply))
+                  }
+            }
       }
     }
   }
 }
 
+// TO DO: Need to convert category selection to MVVM and test
 @Composable
-fun CategorySelection() {
-  LazyColumn {
+fun CategorySelection(uiState: EventsOverviewUiState, modifier: Modifier) {
+  LazyColumn(modifier = modifier) {
     items(EventCategory.entries) { category ->
-      var isChecked by remember { mutableStateOf(false) }
+      var isChecked by remember { mutableStateOf(uiState.categoriesCheckedList.contains(category)) }
       Row(
           verticalAlignment = Alignment.CenterVertically,
           horizontalArrangement = Arrangement.Start,
-          modifier = Modifier.padding(vertical = 0.dp)) {
+          modifier = Modifier.padding(vertical = 0.dp).testTag("categoryRow")) {
             Checkbox(
                 checked = isChecked,
-                onCheckedChange = { isChecked = it },
-                modifier = Modifier.scale(0.6f).size(10.dp).padding(start = 10.dp))
+                onCheckedChange = {
+                  isChecked = it
+                  if (isChecked) {
+                    uiState.categoriesCheckedList.add(category)
```

Comment:
Instead of modifying the categoriesCheckedList directly within the composable function, the UI should call a function in the ViewModel that handles this logic. This will improve our adherence to MVVM principles.

Reply:
Hi Nathaniel, I considered this issue last week too. My approach was to follow what was done for the freeSwitch and RadiusQuery and move the logic into a function in the VM. I kept isChecked the same. This did not work however. Then, I also tried to make isChecked into a Map<EventCategory, Boolean> so we could store the checked state inside EventOverviewUiState. This also did not work. I spent 4 hours last week attempting these approaches and another 2 hours again today. Both attempts have been unsuccessful. 

I have also tried to search online for solutions that use checkbox and follow "MVVM". All solutions seem to use the current working approach I have of tracked isChecked in a mutableStateOf. 

Given that devs are expected to spend 10 hours each week coding, and we have many other features to implement before M3, it may be best to prioritise other features/issues over this one. This will ensure that we continue to deliver value as a whole. 

If you have an other suggested approaches to fix this issue, I am happy for you to try them. Otherwise, we can create a new issue out of this on the SCRUM in our PB.